invokeai/app/invocations/prompt.py

```diff
@@ -55,3 +56,35 @@ def invoke(self, context: InvocationContext) -> PromptCollectionOutput:
             prompts = generator.generate(self.prompt, num_images=self.max_prompts)
 
         return PromptCollectionOutput(prompt_collection=prompts, count=len(prompts))
+    
+
+class PromptsFromFileInvocation(BaseInvocation):
+    '''Loads prompts from a text file'''
+    # fmt: off
+    type: Literal['prompt_from_file'] = 'prompt_from_file'
+
+    # Inputs
+    filename: str = Field(default=None, description="Filename of prompt text file")
+    pre_prompt: str = Field(default=None, description="Add to start of prompt")
+    post_prompt: str = Field(default=None, description="Add to end of prompt")
+    start_line: int = Field(default=1, ge=1, description="Line in the file start start from")
+    max_prompts: int = Field(default=0, ge=0, description="Max lines to read from file (0=all)")
+    #fmt: on
+
+    def promptsFromFile(self, filename: str, pre_prompt: str, post_prompt: str, start_line: int, max_prompts:int):
+        prompts = []
+        start_line -= 1
+        end_line = start_line + max_prompts
+        if max_prompts <= 0:
+            end_line = np.iinfo(np.int32).max
+        with open(filename) as f:
+            for i, line in enumerate(f):
+                if i >= start_line and i < end_line:
+                    prompts.append(pre_prompt + line.strip() + post_prompt)
```

Comment:
There's a potential error here - you cannot add `None` and `str` together. So if `pre_prompt` or `post_prompt` are `None`, this could cause problems. This will need just a bit more logic to safely build up the string to be appended to `prompts`.